test/framework/src/main/java/org/elasticsearch/test/BackgroundIndexer.java

```diff
@@ -157,6 +161,7 @@ public void run() {
                                 }
                                 BulkResponse bulkResponse = bulkRequest.get();
                                 for (BulkItemResponse bulkItemResponse : bulkResponse) {
+                                    ids.add(bulkItemResponse.getId());
```

Comment:
nit: can we move this under the if? it's confusing now as we add the id but don't increment indexCounter